### PR TITLE
Simple NIO tests

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
Make sure we find the provider and it recognizes paths as expected.
These tests don't actually connect to any Google Cloud Storage
bucket.

### CODE REVIEW

File: google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/NIOTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 Google LLC
```
Comment (elharo) [inline google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/NIOTest.java]: 2020

File: google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/NIOTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+@RunWith(JUnit4.class)
+public class NIOTest {
+
+  /** We can create a Path object from a gs:// URI. **/
+  @Test
+  public void testCreatePath() {
+    URI uriToCloudStorage = URI.create("gs://bucket/file.txt");
+    // Return value ignored on purpose, we just want to check
+    // no exception is thrown.
+    Paths.get(uriToCloudStorage);
```
Comment (elharo) [inline google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/NIOTest.java]: at the least, assert this is not null

File: google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/NIOTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    URI uriToCloudStorage = URI.create("gs://bucket/file.txt");
+    Path path = Paths.get(uriToCloudStorage);
+    assertThat(path.getFileSystem().provider().getScheme()).isEqualTo("gs");
+  }
+
+  /** The created Path object has the correct provider class. **/
+  @Test
+  public void testCreatedPathIsCorrectInstance() {
+    URI uriToCloudStorage = URI.create("gs://bucket/file.txt");
+    Path path = Paths.get(uriToCloudStorage);
```
Comment (elharo) [inline google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/NIOTest.java]: You use this setup three times so it should probably be a fixture.

File: google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/NIOTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    assertThat(path.getFileSystem().provider())
+        .isInstanceOf(CloudStorageFileSystemProvider.class);
```
Comment (elharo) [inline google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/NIOTest.java]: We might not want to assert this. Is this part of the published API in some way? Would it matter if this were a different class?

**Comment by jean-philippe-martin** [comment]
Does it still make sense to add this test here, given that the NIO library moved to https://github.com/googleapis/java-storage-nio ?

**Comment by elharo** [comment]
now in a separate repo

**Pull request closed without merging**